sysml.library/Domain Libraries/Geometry/ShapeItems.sysml

```diff
@@ -41,245 +48,716 @@ package ShapeItems {
+		item bfe  [2]	 :> edges;
+		item bre  [2]	 :> edges;
+		item bsle [2]	 :> edges;
+		item bsre [2]	 :> edges;
+		item ufle [2]	 :> edges;
+		item ufre [0..2] :> edges;
+		item urle [2]	 :> edges;
+		item urre [0..2] :> edges;
+
+		assert constraint { ( isEmpty(srf) implies isEmpty(tsre) ) and
+				    ( isEmpty(tsre) == isEmpty(ufre) ) and
+				    ( isEmpty(ufre) == isEmpty(urre) ) }
+
+		item :>> vertices;
+		assert constraint { size(vertices) == size(edges) }
+
+		item tflv [3]	 :> vertices;
+		item tfrv [0..3] :> vertices;
+		item trlv [3]	 :> vertices;
+		item trrv [0..3] :> vertices;
+		item bflv [3]	 :> vertices;
+		item bfrv [3]	 :> vertices;
+		item brlv [3]	 :> vertices;
+		item brrv [3]	 :> vertices;
+		
+		assert constraint { ( isEmpty(tfrv) == isEmpty(trrv) ) }
+
+		/* Bind face edges to specific edges */
+		connection : SelfLink [1] connect tf.edges [0..1] to tfe [0..1];
+		connection : SelfLink [1] connect tf.edges [0..1] to tre [0..1];
+		connection : SelfLink [1] connect tf.edges [0..1] to tsle [0..1];
+		connection : SelfLink [1] connect bf.edges [0..1] to bfe [0..1];
+		connection : SelfLink [1] connect bf.edges [0..1] to bre [0..1];
+		connection : SelfLink [1] connect bf.edges [0..1] to bsle [0..1];
+		connection : SelfLink [1] connect bf.edges [0..1] to bsre [0..1];
+
+		connection : SelfLink [1] connect ff.edges [0..1] to tfe [0..1];
+		connection : SelfLink [1] connect ff.edges [0..1] to bfe [0..1];
+		connection : SelfLink [1] connect ff.edges [0..1] to ufle [0..1];
+
+		connection : SelfLink [1] connect rf.edges [0..1] to tre [0..1];
+		connection : SelfLink [1] connect rf.edges [0..1] to bre [0..1];
+		connection : SelfLink [1] connect rf.edges [0..1] to urle [0..1];
+
+		/* connection : SelfLink edge connect vertices to specific vertices */
+
+		connection : SelfLink [1] connect tfe.vertices [0..1] to tflv [0..1];
+		connection : SelfLink [1] connect tre.vertices [0..1] to trlv [0..1];
+		connection : SelfLink [1] connect tsle.vertices [0..1] to tflv [0..1];
+		connection : SelfLink [1] connect tsle.vertices [0..1] to trlv [0..1];
+
+		connection : SelfLink [1] connect bfe.vertices [0..1] to bflv [0..1];
+		connection : SelfLink [1] connect bfe.vertices [0..1] to bfrv [0..1];
+		connection : SelfLink [1] connect bre.vertices [0..1] to brlv [0..1];
+		connection : SelfLink [1] connect bre.vertices [0..1] to brrv [0..1];
+		connection : SelfLink [1] connect bsle.vertices [0..1] to bflv [0..1];
+		connection : SelfLink [1] connect bsle.vertices [0..1] to brlv [0..1];
+		connection : SelfLink [1] connect bsre.vertices [0..1] to bfrv [0..1];
+		connection : SelfLink [1] connect bsre.vertices [0..1] to brrv [0..1];
+
+		connection : SelfLink [1] connect ufle.vertices [0..1] to tflv [0..1];
+		connection : SelfLink [1] connect ufle.vertices [0..1] to bflv [0..1];
+		connection : SelfLink [1] connect urle.vertices [0..1] to trlv [0..1];
+		connection : SelfLink [1] connect urle.vertices [0..1] to brlv [0..1];
```

Comment:
Instead of `connection : SelfLink [1] connect ... to ...;` these should be `binding [1] bind ... = ...;`.

Reply:
Thx, didn't know about the binding keyword.